File: agrario-front/src/app/modelos/AsignacionServicio.ts

import { Cliente } from "./Cliente";
import { Servicio } from "./Servicio";
export class AsignacionServicio {
  constructor(
    public idAsignacion: number | null,
    public servicio: number,        // ID del servicio
    public cliente: number,         // ID del cliente
    public metodo: string,
    public fechaAsignacion: string  // Formato ISO
  ) {}
}


export class AsignacionServicioReport {
  constructor(
    public idAsignacion: number | null,
    public servicio: Servicio,        // Objeto Servicio completo
    public cliente: Cliente,          // Objeto Cliente completo
    public metodo: string,
    public fechaAsignacion: string
  ) {}
}

File: Agrario/src/main/java/pe/proyecto/agrario/agrario/modelo/AsignacionServicio.java
package pe.proyecto.agrario.agrario.modelo;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;



@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "asignacion_servicio")
public class AsignacionServicio {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_asignacion")
    private Long idAsignacion;

    @ManyToOne
    @JoinColumn(name = "id_servicio", nullable = false)
    private Servicio servicio;

    @ManyToOne
    @JoinColumn(name = "id_cliente", nullable = false)
    private Cliente cliente;

    private String metodo;

    private LocalDate fechaAsignacion;

}
